### DISCUSSION TITLE
Anybody can make this script work?

### DISCUSSION THREAD

**Discussion by surferHalo** [JetBrains/ideavim#451] [Q&A] [answered]
Recently, ideavim being able support the vimscript, I know nothing about vimscript, but I have a requirements that is discussed and solved in below links:
https://vi.stackexchange.com/questions/12026/ammend-w-behavior-ignore-operators-and-brackets
https://stackoverflow.com/questions/6527698/setting-to-skip-over-punctuation-when-moving-forwards-and-backwards-words/6528201

I think the sidyll's solution is the most simple one, but I can not make it work in ideavim.

```
function! JumpToNextWord()
    normal w
    while strpart(getline('.'), col('.')-1, 1) !~ '\w'
        normal w
    endwhile
endfunction

nnoremap <silent> ,w :call JumpToNextWord()<CR>
```

Anybody help? Thanks in advance!

**Comment by AlexPl292** [comment, marked as answer]
Hi, unfortunately vimscript support in IdeaVim is on it's early stages and it doesn't support yet some functions from your script like `getline` or `strpart`.

**Reply by surferHalo** [reply]
Hi, Alex, Is there any function can get the character under the cursor in idea vimscript? if so, I think this problem can be easily solved.

**Reply by AlexPl292** [reply]
No, as I remember, we literally support at the moment only `sin` function as an example. We plan to start implementing the vim functions, but this would take some time and there is a small refactoring inside that should be done before starting that.

If you fancy, there is an ability to create a regular IJ plugin that depends on IdeaVim and implements an extension for IdeaVim (like [ideavim-sneak](https://github.com/Mishkun/ideavim-sneak)), but this is of course not that simple as put a few lines into `~/.ideavimrc` file.
